mkosi/__init__.py

```diff
@@ -6285,6 +6286,51 @@ def run_ssh(args: CommandLineArguments) -> None:
         )
 
 
+def generate_secure_boot_key(args: CommandLineArguments) -> NoReturn:
+    """Generate secure boot keys using openssl"""
+    args.secure_boot_key = args.secure_boot_key or "./mkosi.secure-boot.key"
+    args.secure_boot_certificate = args.secure_boot_certificate or "./mkosi.secure-boot.crt"
+
+    keylength = 2048
+    expire_in_days = 2 * 365
+    expiration_date = datetime.date.today() + datetime.timedelta(expire_in_days)
+    user = os.getenv("SUDO_USER") or os.getenv("USER") or "unknown user"
+    cn = f"mkosi of {user}"
+
+    for f in (args.secure_boot_key, args.secure_boot_certificate):
+        if os.path.exists(f):
+            die(
+                f"{f} already exists.\n"
+                "If you are sure you want to generate new secure boot keys\n"
+                f"remove {args.secure_boot_key} and {args.secure_boot_certificate} first."
+            )
```

Comment:
this check should probably be skipped if --force is specified (in which case the key files should just be replaced)